File: cpp/02-gtest-ut/test/FizzBuzz2Test.cpp
#include <gtest/gtest.h>
#include "../src/FizzBuzz2.hpp"

class FizzBuzz2Test : public ::testing::Test {
protected:
    FizzBuzz2 sut;
};

TEST_F(FizzBuzz2Test, when_1_then_1) {
    EXPECT_EQ("1", sut.fizzBuzz(1));
}

TEST_F(FizzBuzz2Test, _3_then_fizz) {
    EXPECT_EQ("Fizz", sut.fizzBuzz(3));
}

TEST_F(FizzBuzz2Test, _5_then_buzz) {
    EXPECT_EQ("Buzz", sut.fizzBuzz(5));
}

TEST_F(FizzBuzz2Test, _15_then_fizzbuzz) {
    EXPECT_EQ("FizzBuzz", sut.fizzBuzz(15));
}

File: javascript/02-mocha/test/fizzbuzz2.test.js
const { expect } = require('chai');
const { fizzBuzz } = require('../src/fizzbuzz2');

describe('fizzBuzz', () => {
  it('returns 1 for 1', () => {
    expect(fizzBuzz(1)).to.equal('1');
  });

  it('returns Fizz for multiples of 3', () => {
    expect(fizzBuzz(3)).to.equal('Fizz');
  });

  it('returns Buzz for multiples of 5', () => {
    expect(fizzBuzz(5)).to.equal('Buzz');
  });

  it('returns FizzBuzz for multiples of 15', () => {
    expect(fizzBuzz(15)).to.equal('FizzBuzz');
  });
});
